SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs

```diff
@@ -0,0 +1,258 @@
+            }
+
+            _contentSurface.Brush.Stretch = CompositionStretch.UniformToFill;
+            _combinedEffect = new CompositeEffect()
+            {
+                Mode = Microsoft.Graphics.Canvas.CanvasComposite.SourceOut,
+                Sources =
+                {
+                    new CompositionEffectSourceParameter("Noise"),
+                    new CompositionEffectSourceParameter("Content"),
+                }
+            };
+            var combinedEffectFactory = compositor.CreateEffectFactory(_combinedEffect);
+            _combinedEffectBrush = combinedEffectFactory.CreateBrush();
+            _combinedEffectBrush.SetSourceParameter("Noise", _noiseSurface.Brush);
+            _combinedEffectBrush.SetSourceParameter("Content", _contentSurface.Brush);
+
+            _outputVisual = compositor.CreateSpriteVisual();
+            _outputVisual.Brush = _combinedEffectBrush;
+            _outputVisual.Size = EffectSize;
+
+            ElementCompositionPreview.SetElementChildVisual(Placeholder, _outputVisual);
+
+            _initialized = true;
+        }
+
+        private void NoiseDrawHandler(CompositionDrawingSurface surface, CompositionGraphicsDevice device)
+        {
+            if (_noiseLuminanceEffect == null || surface.Size.Width == 0 || surface.Size.Height == 0)
+            {
+                return;
+            }
+            using (var ds = CanvasComposition.CreateDrawingSession(surface))
+            {
+                ds.Clear(Colors.Transparent);
+                ds.DrawImage(_noiseLuminanceEffect);
+            }
+        }
+
+        private void DisposeEffect()
+        {
+            if (_noiseSurface != null)
+            {
+                _noiseSurface.Dispose();
+                _noiseSurface = null;
+            }
+            if (_noiseLuminanceEffect != null)
+            {
+                _noiseLuminanceEffect.Dispose();
+                _noiseLuminanceEffect = null;
+            }
+            if (_noiseEffect != null)
+            {
+                _noiseEffect.Dispose();
+                _noiseEffect = null;
+            }
+            if (_outputVisual != null)
+            {
+                _outputVisual.Dispose();
+                _outputVisual = null;
+            }
+            if (_combinedEffectBrush != null)
+            {
+                _combinedEffectBrush.Dispose();
+                _combinedEffectBrush = null;
+            }
+            if (_combinedEffect != null)
+            {
+                _combinedEffect.Dispose();
+                _combinedEffect = null;
+            }
+            if (_contentSurface != null)
+            {
+                _contentSurface.Dispose();
+                _contentSurface = null;
+            }
+            _initialized = false;
+        }
+
+        private void ResizeEffect()
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            _outputVisual.Size = EffectSize;
```

Comment:
You also need to resize the _noiseEffect in response to size changes.
_noiseEffect.Size = _outputVisual.Size;